test/mongo.test.js

```diff
@@ -44,24 +43,829 @@ describe('mongo tests', function () {
     si.ready(done)
   })
 
-  Shared.basictest({
-    seneca: si,
-    senecaMerge: senecaMerge,
-    script: lab,
+  describe('basic tests', () => {
+    Shared.basictest({
+      seneca: si,
+      senecaMerge: senecaMerge,
+      script: lab
+    })
+  })
+
+  describe('limit tests', () => {
+    Shared.limitstest({
+      seneca: si,
+      script: lab
+    })
   })
 
-  Shared.limitstest({
-    seneca: si,
-    script: lab,
+  describe('sort tests', () => {
+    Shared.sorttest({
+      seneca: si,
+      script: lab
+    })
   })
 
-  Shared.sorttest({
-    seneca: si,
-    script: lab,
+  describe('upsert tests', () => {
+    before(prepareForRaceConditionTesting)
+
+    after(unprepareAfterRaceConditionTesting)
+
+    Shared.upserttest({
+      seneca: si,
+      script: lab
+    })
+
+    // NOTE: WARNING: The reason we need a unique index on the users.email
```

Comment:
seneca-store-test could define this as a "standard" requirement - that's OK
add a note in the seneca-store-test README

Reply:
As requested:
https://github.com/senecajs/seneca-store-test/pull/60/commits/878376b4dbda0e2c15a7d433dff6c66a063aa384
https://github.com/senecajs/seneca-store-test/pull/60/commits/5e9b1c4a907657da743f01e7402af1641ea4bc8d